File: lab-report-api/src/main/java/com/ubo/labreport/model/Report.java
package com.ubo.labreport.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "t_report")
public class Report {
    @Id
    @GeneratedValue(generator = "UUID")
    private UUID id;
    private String patientFirstName;
    private String patientLastName;
    private String identityNumber;
    private String diagnosis;
    private String diagnosisDetails;
    private LocalDateTime givenDate;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    @JoinColumn(name = "laborant_id", nullable = false)
    private Laborant laborant;

    @OneToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "image_id")
    private ImageData image;
}

File: lab-report-api/src/main/java/com/ubo/labreport/model/Report.kt
package com.ubo.labreport.model

import jakarta.persistence.*
import org.hibernate.annotations.GenericGenerator
import java.time.LocalDateTime

@Entity
data class Report(

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    val id: String,
    val patientFirstName: String,
    val patientLastName: String,
    val identityNumber: String,
    val diagnosis: String,
    val diagnosisDetails: String,
    val givenDate: LocalDateTime,

    @ManyToOne(fetch = FetchType.EAGER, cascade = [CascadeType.ALL])
    @JoinColumn(name = "laborant_id", nullable = false)
    val laborant: Laborant

) {


    constructor(patientFirstName: String,patientLastName: String,identityNumber: String,diagnosis: String,
                diagnosisDetails: String,laborant: Laborant) : this(
        "",
        patientFirstName = patientFirstName,
        patientLastName = patientLastName,
        identityNumber = identityNumber,
        diagnosis = diagnosis,
        diagnosisDetails = diagnosisDetails,
        LocalDateTime.now(),
        laborant = laborant
    )

    constructor(id: String,patientFirstName: String,patientLastName: String,identityNumber: String,diagnosis: String,
                diagnosisDetails: String,laborant: Laborant) : this(
        "",
        patientFirstName = patientFirstName,
        patientLastName = patientLastName,
        identityNumber = identityNumber,
        diagnosis = diagnosis,
        diagnosisDetails = diagnosisDetails,
        LocalDateTime.now(),
        laborant = laborant
    )


}

